st_weaviate_connection/connection.py

```diff
@@ -1,63 +1,204 @@
 
-    def _create_auth_config(self) -> Optional[weaviate.AuthApiKey]:
+    def _create_auth_config(self) -> Optional[_APIKey]:
         api_key = self.api_key or self._secrets.get("WEAVIATE_API_KEY")
         if api_key is not None:
-            return weaviate.AuthApiKey(api_key=api_key)
+            return Auth.api_key(api_key=api_key)
         else:
             return None
 
-    def _convert_to_dataframe(self, results) -> pd.DataFrame:
-        class_name = list(results["data"]["Get"].keys())[0]
-        data = results["data"]["Get"][class_name]
-        df = pd.json_normalize(data)
+    def _gql_to_dataframe(self, results: _RawGQLReturn) -> pd.DataFrame:
+        collection_name = list(results.get.keys())[0]
+        data = results.get[collection_name]
+        df = pd.DataFrame(data)
         return df
 
-    def query(self, query: str, ttl: int = 3600, **kwargs) -> pd.DataFrame:
-        @cache_data(ttl=ttl)
-        def _query(query: str, **kwargs):
-            results = self._connect().query.raw(query)
-            if "errors" in results:
-                error_message = (
-                    f"The GraphQL query returned an error: {results['errors']}"
-                )
+    def hybrid_search(
+        self,
+        collection_name: str,
+        query: str,
+        limit: int = 10,
+        filters: Optional[_Filters] = None,
+        target_vectors: Optional[TargetVectorJoinType] = None,
+        query_properties: Optional[List[str]] = None,
+        alpha: float = 0.7,
+    ) -> pd.DataFrame:
+        """
+        Query a Weaviate collection using a simplified hybrid query.
+
+        Parameters
+        ----------
+        collection_name : str
+            The name of the collection to query.
+        query : str
+            The query to search for.
+        limit : int, optional
+            The number of results to return.
+            Default: 10.
+        filters : Filter, optional
+            The filters to apply to the query.
+            Import the `WeaviateFilter` class from `st_weaviate_connection` to use this argument.
+        target_vectors : string, List[string], List[TargetVectors], optional
+            The target vector(s) to search in the semantic search part of the query.
+            Only required if the target collection uses named vectors.
+            Optionally, import the `WeaviateTargetVectors` class from `st_weaviate_connection` to use this argument.
+        query_properties : List[str], optional
+            The properties to query in the keyword search part of the query.
+            If not provided, all properties are queried.
+        alpha: float, optional
+            The weight of the semantic search part of the query. (alpha=1 is a semantic search, alpha=0 is a keyword search).
+            If not provided, the Weaviate server default value is used.
+        """
+
+        collection = self._client.collections.get(name=collection_name)
+
+        def _hybrid_search(
```

Comment:
Why don't we annotate `_hybird_search` with @cache_data(ttl=ttl) ?

Reply:
When I tried it earlier, I found that some of the parameters could not be hashed, so the cache would not work sadly.

Reply:
Then I think we should document that in the README or notebook, since caching is a common technique used by streamlit connectors and so users probably expect that by default

Reply:
Updated!